### PR TITLE
OpenCL: Time of feature collection as in- and output

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

File: operators/src/opencl/cl_program.rs [hunk 1]
```diff
@@ -2063,6 +2098,79 @@ __kernel void gid(
...
+        ));
+
+        let mut compiled = cl_program.compile(kernel, "points").unwrap();
+
+        let mut runnable = compiled.runnable();
+        runnable.set_input_features(0, &typed_input).unwrap();
+        runnable.set_output_features(0, &mut out).unwrap();
+        compiled.run(runnable).unwrap();
+
+        let result = out.output.unwrap().get_points().unwrap();
```
Comment (michaelmattig) [inline operators/src/opencl/cl_program.rs:2169]: please check if time is actually set correctly

File: datatypes/src/collections/batch_builder.rs [hunk 2]
```diff
@@ -147,7 +147,12 @@ impl RawFeatureCollectionBuilder {
     pub fn set_time_intervals(&mut self, values_buffer: Buffer) -> Result<()> {
         let data = ArrayData::builder(TimeInterval::arrow_data_type())
             .len(self.num_features)
-            .add_buffer(values_buffer)
+            .add_child_data(
```
Comment (michaelmattig) [inline datatypes/src/collections/batch_builder.rs:150]: please add test that uses this method

**Comment by michaelmattig** [comment]
looks good. please add test and we can merge it

**Review by michaelmattig — CHANGES_REQUESTED** [review]

File: operators/src/opencl/cl_program.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply operators/src/opencl/cl_program.rs:2169]: Here, I check if the result equals the input, i.e., it needs to have equal time intervals.

File: datatypes/src/collections/batch_builder.rs (hunk 2 above)
Comment (ChristianBeilschmidt) [reply datatypes/src/collections/batch_builder.rs:150]: 5b87247

**Pull request merged**